lib/filter/filter-netmask6.c

```diff
@@ -114,42 +110,80 @@ _eval(FilterExprNode *s, LogMessage **msgs, gint num_msg)
     }
 
   memset(&network_address, 0, sizeof(struct in6_addr));
-  get_network_address((unsigned char *) &address, self->prefix, &network_address);
-  result = _in6_addr_compare(&network_address, &self->address);
+  filter_netmask6_calc_network_address((unsigned char *) &address, self->prefix, &network_address);
+  result = _in6_addr_equals(&network_address, &self->address);
 
   return result ^ s->comp;
 }
 
-FilterExprNode *
-filter_netmask6_new(gchar *cidr)
+static void
+_get_validated_address(FilterNetmask6 *self, gchar *cidr, gchar address[])
```

Comment:
This should be splitted into two different functions for it has multiple responsibilities. 
* for normalizing the key: `normalize_address`
* for the settings the defaults if address cannot be normalized: `set_address_defaults()`